runtime/util/thrinfo.c

```diff
@@ -420,8 +420,9 @@ getVMThreadStateHelper(J9VMThread *targetThread,
 								lockOwnerObject = J9OBJECT_FROM_FJ9OBJECT(targetThread->javaVM, fobj);
 							}
 #else
+							/* Pass in the current thread as the macro expects the currently executing thread */
 							lockOwnerObject =
-								J9VMJAVAUTILCONCURRENTLOCKSABSTRACTOWNABLESYNCHRONIZER_EXCLUSIVEOWNERTHREAD(targetThread, lockObject);
+								J9VMJAVAUTILCONCURRENTLOCKSABSTRACTOWNABLESYNCHRONIZER_EXCLUSIVEOWNERTHREAD(targetThread->javaVM->internalVMFunctions->currentVMThread(targetThread->javaVM), lockObject);
```

Comment:
Actually, it's simpler just to use _VM variant of the same API: J9VMJAVAUTILCONCURRENTLOCKSABSTRACTOWNABLESYNCHRONIZER_EXCLUSIVEOWNERTHREAD_VM
and supply javaVM instead of targetThread

Reply:
while we did not see a problem, I suspect we need a similar treatment for J9VMJAVALANGTHREAD_THREADREF, just a few lines below.

Bottom line, any macro/API that may triggering an object access barrier, has to supply the current running thread, or if it does not have it easily accessible use _VM variant of the API.